scaling.tex

```diff
@@ -1 +1,216 @@
-\section{\label{sec:scale}Scaling Analysis}
\ No newline at end of file
+\section{\label{sec:scale}Scaling Analysis}
+The Gung Ho dynamical core uses an unstructured mesh to avoid a
+singular pole which ultimately inhibits the scaling of the
+UM. Examining the scaling behaviour of the model to see if it
+performs sufficiently well is therefore important. To test the scaling a series
+of model runs was performed. The version of the LFRic repository trunk
+was $17483$. The code was compiled on the XCS with the Intel 17
+compiler, at production level which is \verb+-O3+. The baroclinic wave
+test was run on a $C576$ mesh. This roughly corresponds to a $17$km
+horizontal resolution. The model was run with a $30$km lid and with
+$30$ levels.  The time-step used was $180$s
+seconds and the benchmarks were run for $100$ time-steps. The code was
+profiled using the CrayPAT tool in sampling mode. 
+
+\begin{figure}
+\centering\includegraphics[width=1.0\linewidth]{figs/wc-scale.eps}
+\caption{\label{fig:wc_scale}Wall-clock time strong scaling of the 
+  Baroclinic wave test. In red, labelled MPI is an MPI only
+  version. In yellow, labelled OMP is the hybrid MPI+OpenMP version.}
+\end{figure} 
+
+Shown in figure~\ref{fig:wc_scale} is the strong scaling of the whole
+model up to $384$ nodes. The MPI only version ran with $36$ MPI ranks
+per node. The hybrid version ran with $6$ MPI ranks per node and $6$
+openMP threads per MPI rank. The MPI only version on $384$ nodes
+crashed during the partitioning phase with an Out Of Memory (OOM)
+error. The global mesh is read in by every MPI rank and then
+partitioned. This can cause memory problems for large meshes on larger
+node counts. This can be solved by a single rank per node reading in
```

Comment:
"This can be solved"... does that mean we implemented the single rank per node reading?

Reply:
No, it means we should though.  Added a footnote.